src/frontend/components/SchedulingFields/index.tsx

```diff
@@ -0,0 +1,231 @@
+import { Box, DateInput, FormField, TextInput } from 'grommet';
+import { MarginType } from 'grommet/utils';
+import { DateTime, Duration, Settings } from 'luxon';
+import React from 'react';
+import { defineMessages, useIntl } from 'react-intl';
+
+import { Nullable } from 'utils/types';
+
+const messages = defineMessages({
+  startingAtDateTextLabel: {
+    defaultMessage: 'Starting date',
+    description:
+      'Label of the input field to add/update the start date of the scheduled event',
+    id: 'component.SchedulingFields.startingAtDateTextLabel',
+  },
+  startingAtTimeTextLabel: {
+    defaultMessage: 'Starting time',
+    description:
+      'Label of the input field to add/update the start time of the scheduled event',
+    id: 'component.SchedulingFields.startingAtTextLabel',
+  },
+  estimatedDurationTextLabel: {
+    defaultMessage: 'Estimated duration',
+    description:
+      'Label of the input field to add/update the estimated duration of the scheduled event',
+    id: 'component.SchedulingFields.estimatedDurationTextLabel',
+  },
+});
+
+// build a list of times from 09h00 to 19h30
+const currentTimeForSuggestions = Duration.fromObject({ hours: 9 });
+const allTimesSuggestions = Array.from({ length: 22 }, (_v, k) => k * 30).map(
+  (minutes) => currentTimeForSuggestions.plus({ minutes }).toFormat('hh:mm'),
+);
+
+const displayedDuration = (duration: Duration) => {
+  return duration.toFormat('h:mm');
+};
+
+const returnedDuration = (durationString: string) => {
+  if (durationString.length === 4) durationString = '0' + durationString;
+  return durationString
+    ? Duration.fromISOTime(durationString).toFormat('hh:mm:ss')
+    : null;
+};
+
+// build a list of durations from 15 minutes to 5 hours
```

Comment:
you can expand to 12 hours. Maybe in that case the step should not be 15 minutes but 30.

Reply:
Same comment as above, I'll push, and we'll decide.

Reply:
It's pushed.